src/Aspire.Hosting/ApplicationModel/DeploymentImageTagAnnotation.cs

```diff
@@ -6,19 +6,51 @@
 
 namespace Aspire.Hosting.ApplicationModel;
 
+/// <summary>
+/// Context information for deployment image tag callback functions.
+/// </summary>
+[Experimental("ASPIRECOMPUTE001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+public sealed class DeploymentImageTagCallbackAnnotationContext
```

Comment:
Add a `DistributedApplicationExecutionContext ExecutionContext` property this class.